ODIN_II/SRC/include/HardSoftLogicMixer.hpp

```diff
@@ -0,0 +1,127 @@
+/*
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following
+ * conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef HARD_SOFT_LOGIC_MIXER_HPP
+#define HARD_SOFT_LOGIC_MIXER_HPP
+
+#include "odin_types.h" // mix_hard_blocks, config_t
+
+/**
+ * @brief Weighted node
+ * WeightedNode.node is a pointer to nnode_t (the candidate for optimization)
+ * WeightedNode.weight is a field that is used for storing weights towards the 
+ * mixing optimization.
+ *  value of -1 is reserved for hardened blocks
+ *  value 0 is the default value
+ */
+typedef struct {
+    nnode_t* node;
+    long weight;
+} WeightedNode;
```

Comment:
I like it, but I still think it can be useful in the longer run for other thing, having it baked into the node can allow the weight to be accounted for in future optimization, and be used by the netlist_stats stuff

Reply:
Thank you for your review, I've updated the code according to your suggestions